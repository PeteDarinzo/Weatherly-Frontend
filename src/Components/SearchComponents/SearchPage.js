import React from "react";
import MovieSearchForm from "./MovieSearchForm";
import SearchList from "./SearchList";
import Grid from '@mui/material/Grid';
import Container from '@mui/material/Container';

const SearchPage = ({ getMovies, searchResults, saveMovie }) => {

  return (
    <Container>
      <Grid container>
        <Grid item md={6}>
          <MovieSearchForm getMovies={getMovies} />
        </Grid>
        <Grid item md={6}>
          <SearchList movies={searchResults} saveMovie={saveMovie} />
        </Grid>
      </Grid>
    </Container>
  );
}

export default SearchPage;