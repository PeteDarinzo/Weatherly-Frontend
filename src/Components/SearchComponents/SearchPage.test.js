import SearchPage from "./SearchPage";
import { render } from '@testing-library/react';
import { MemoryRouter} from "react-router-dom";

it("renders a search page without crashing", () => {
  render(
    <MemoryRouter>
      <SearchPage getMovies={() => { }} searchResults={[]} saveMovie={() => { }} />
    </MemoryRouter>
  );
});


it("matches the search page snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchPage getMovies={() => { }} searchResults={[]} saveMovie={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
