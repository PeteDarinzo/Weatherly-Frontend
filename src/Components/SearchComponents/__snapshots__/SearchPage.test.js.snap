// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the search page snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-1k7sk4d-MuiGrid-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="makeStyles-form-4 MuiBox-root css-0"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              Search Titles
            </h4>
            <form>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-5 css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="title"
                    id="title-label"
                  >
                    Title
                    <span
                      aria-hidden="true"
                      class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="title"
                      name="title"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1g8cwq9"
                      >
                        <span>
                          Title *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jh47zj-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Search
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-1k7sk4d-MuiGrid-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm makeStyles-container-6 css-cuefkz-MuiContainer-root"
        >
          <div
            class="css-nen11g-MuiStack-root"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
